### DISCUSSION TITLE
luasnip.extras.select_choice: stay in insert mode after selection is made?

### DISCUSSION THREAD

**Discussion by pogopaule** [L3MON4D3/LuaSnip#497] [Q&A] [answered]
The select_choice popup is awesome! The only problem I have is that vim is in normal mode after I have made a selection, which breaks the flow if there are more nodes to go to.
How can I stay in insert mode?

**Comment by wassimk** [comment]
I have the same question! Often times I have more jumps in the snippet and don't want to exit insert mode after UI select. It also causes a problem where my [ghost text saying it's a choice](https://github.com/wassimk/dotfiles/blob/642870e46e63180dc440534e5e5e3e9f7bf85874/config/nvim/lua/wassim/luasnip.lua#L12) stays until I go back into insert mode and exit again.

I was looking at the [`select_choice`](https://github.com/L3MON4D3/LuaSnip/blob/50760912f54dfd0087b29abef6ede39c33615ab9/lua/luasnip/extras/select_choice.lua#L12-L21) method, and it doesn't look like we can do it as it's written. It would be cool if we could pass a function for it to execute. We could call `vim.cmd('startinsert')` in that function

**Comment by L3MON4D3** [comment, marked as answer]
Ouh, good point, I totally agree :+1:
I pushed a (potential: it works for me, but I don't know about `vim.ui.select`-providers other than dressing with telescope) fix to the `select_choice_reinsert_test`-branch, could you try if it's reliable?

**Reply by wassimk** [reply]
@L3MON4D3 it works for me! I am using Telescope.

**Reply by pogopaule** [reply]
@L3MON4D3 works for me as well. :+1:  I also use Telescope (via https://github.com/nvim-telescope/telescope-ui-select.nvim)

**Reply by L3MON4D3** [reply]
Okay, I've pushed it to master for now
